librosa/feature/spectral.py

```diff
@@ -352,16 +365,20 @@ def spectral_bandwidth(
     if freq is None:
         freq = fft_frequencies(sr=sr, n_fft=n_fft)
 
+    # TODO: shape test freq
+
+
     if freq.ndim == 1:
-        deviation = np.abs(np.subtract.outer(freq, centroid[0]))
+        deviation = np.abs(np.subtract.outer( centroid[...,0,:],freq).swapaxes(-2,-1))
     else:
-        deviation = np.abs(freq - centroid[0])
+        deviation = np.abs(freq - centroid)
+        
 
     # Column-normalize S
     if norm:
-        S = util.normalize(S, norm=1, axis=0)
+        S = util.normalize(S, norm=1, axis=-2)
 
-    return np.sum(S * deviation ** p, axis=0, keepdims=True) ** (1.0 / p)
+    return np.sum(S * deviation ** p, axis=-2, keepdims=True) ** (1.0 / p)
```

Comment:
Bandwidth LGTM; that broadcast subtraction still feels a little awkward to me, but I can live with it for now.